SwiftUIDZ/SwiftUIDZ/CustomTabBarView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  CustomTabBarView.swift
+//  SwiftUIDZ
+//
+//  Created by Анна Сычева on 21.01.2023.
+//
+
+import SwiftUI
+
+/// Кастомный таббар
+struct CustomTabBarView: View {
+    
+    // MARK: - Constants
+    
+    private enum Constants {
+        static let duration = 0.2
+        static let imageSize: CGFloat = 30
+        static let circleSize: CGFloat = 25
+        static let offset: CGFloat = 40
+        static let paddingBottom: CGFloat = 10
+        static let offsetMultiplay = 1.02
+    }
+    
+    // MARK: - Public Properties
+    
+    @Binding var currentTab: TabType
+    
+    // MARK: - Body
+    
+    var body: some View {
+        tabIconWithCircle
+    }
+    
+    // MARK: - Private Properties
+    
+    private var tabIconWithCircle: some View {
+        GeometryReader { proxy in
+            let width = proxy.size.width
+            HStack {
+                ForEach(TabType.allCases, id: \.rawValue) { tab in
+                    Button {
+                        withAnimation(.easeInOut(duration: Constants.duration)) {
+                            currentTab = tab
+                        }
+                    } label: {
+                        Image(systemName: tab.rawValue)
+                            .renderingMode(.template)
+                            .resizable()
+                            .aspectRatio(contentMode: .fit)
+                            .frame(width: Constants.imageSize, height: Constants.imageSize)
+                            .frame(maxWidth: .infinity)
+                            .foregroundColor(currentTab == tab ? .blue : .gray)
+                    }
+                }
+            }
+            .frame(maxWidth: .infinity)
+            .background(alignment: .leading) {
+                Circle()
+                    .fill(Color.yellow)
+                    .frame(width: Constants.circleSize, height: Constants.circleSize)
+                    .offset(x: Constants.offset)
+                    .offset(x: indicatorOffset(width: width * Constants.offsetMultiplay))
+            }
+        }
+        .frame(height: Constants.imageSize)
+        .padding(.bottom, Constants.paddingBottom)
+        .padding([.horizontal, .top])
+    }
+    
+    // MARK: - Private Methods
+    
+    private func indicatorOffset(width: CGFloat) -> CGFloat {
+        let index = CGFloat(getIndex())
+        if index == 0 {
+            return 0
+        }
+        let buttonWidth = width / CGFloat(TabType.allCases.count)
+        return index * buttonWidth
+    }
+    
+    private func getIndex() -> Int {
```

Comment:
```suggestion
    private func getIndex() -> Int {
```
это должно быть во viewModel